test/core.js

```diff
@@ -595,5 +595,28 @@ test('resolve(specifier, base?, conditions?)', async function (t) {
     process.emitWarning = oldEmitWarning
   })()
 
+  t.is(
+    await resolve(
+      'package-self-import-1',
+      new URL('node_modules/package-self-import-1/', import.meta.url).href
+    ),
+    new URL('node_modules/package-self-import-1/index.js', import.meta.url)
+      .href,
+    'should be able to resolve a self-import'
+  )
+
+  t.is(
+    await resolve(
+      'package-self-import-1',
+      new URL(
+        'node_modules/package-self-import-1/test/index.js',
+        import.meta.url
+      ).href
+    ),
+    new URL('node_modules/package-self-import-1/index.js', import.meta.url)
+      .href,
+    'should be able to resolve a self-import from a sub-file'
+  )
```

Comment:
Formatting is off, run `npm test` to check everything

Reply:
hm... I am not getting any formatting errors...

Reply:
Correct, it automatically fixes things for you!